[aotinductor] Make writing of the weight files to be conditional

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111379

Summary: Since we cache the AOTInductor generated library file, we should not need to write the weights as binary file if the library file already exists.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

desertfire committed:
[aotinductor] Make writing of the weight files to be conditional

Summary: Since we cache the AOTInductor generated library file, we should not need to write the weights as binary file if the library file already exists.

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/codecache.py:
```diff
@@ -1369,6 +1347,29 @@ def _to_bytes(t: torch.Tensor) -> bytes:
+
+                        return bytes(raw_array.contents)
+
+                    aot_constants = b""
+                    for tensor in graph.constants.values():
+                        aot_constants += _to_bytes(tensor)
```
```suggestion
                    aot_constants = b"".join(_to_bytes(tensor) for tensor in graph.constants.values())
```

desertfire committed:
Update torch/_inductor/codecache.py

Co-authored-by: Aaron Gokaslan <aaronGokaslan@gmail.com>

desertfire committed:
Update on "[aotinductor] Make writing of the weight files to be conditional"


Summary: Since we cache the AOTInductor generated library file, we should not need to write the weights as binary file if the library file already exists.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "Previously inductor tests passed and this one is lint fix only"
